### PR TITLE
12304103-ali

### PR DESCRIPTION
**Pull request by abdoulkaled**
12304103-ali

### CODE REVIEW

File: ok [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (abdoulkaled) [inline ok:1]: <resources>

```
<!-- Base application theme. -->
<style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
    <!-- Customize your theme here. -->
</style>
```

</resources>

Comment (abdoulkaled) [inline ok:1]: <resources>
    
    <dimen name="activity_horizontal_margin">16dp</dimen>
    <dimen name="activity_vertical_margin">16dp</dimen>
    <dimen name="textview_padding">8dp</dimen>
    <dimen name="textview_margin">8dp</dimen>
</resources>

Comment (abdoulkaled) [inline ok:1]: <?xml version="1.0" encoding="utf-8"?>
<resources>

<string name="app_name">Tip Calculator</string>
<string name="amount">Amount:</string>
<string name="fifteen_percent">15%</string>
<string name="eighteen_percent">18%</string>
<string name="custom_tip_percentage">Custom %</string>
<string name="tip">Total Tip</string>
<string name="total">Total per person</string>
<string name="partySize">number in party</string>
<string name="number">Number in party:</string>
    <string name="action_settings">action</string>

</resources>

Comment (abdoulkaled) [inline ok:1]: <menu xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools" tools:context=".MainActivity">
    <item android:id="@+id/action_settings" android:title="@string/action_settings"
        android:orderInCategory="100" app:showAsAction="never" />
</menu>

Comment (abdoulkaled) [inline ok:1]: <GridLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/gridLayout"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:columnCount="2"
    android:orientation="horizontal"
    android:paddingBottom="@dimen/activity_vertical_margin"
    android:paddingLeft="@dimen/activity_horizontal_margin"
    android:paddingRight="@dimen/activity_horizontal_margin"
    android:paddingTop="@dimen/activity_vertical_margin"
    android:useDefaultMargins="true"
    tools:context=".MainActivity" >

```
<TextView
    android:id="@+id/amountTextView"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_gravity="right|center_vertical"
    android:labelFor="@+id/amountEditText"
    android:text="@string/amount"
    android:textAppearance="?android:attr/textAppearanceMedium" />

<EditText
    android:id="@+id/amountEditText"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_row="0"
    android:digits="0123456789"
    android:inputType="number"
    android:maxLength="6" >

    <requestFocus />
</EditText>

<TextView
    android:id="@+id/amountDisplayTextView"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_gravity="fill_horizontal"
    android:layout_row="0"
    android:background="@android:color/holo_blue_bright"
    android:padding="@dimen/textview_padding"
    android:textAppearance="?android:attr/textAppearanceMedium" />


<TextView
    android:id="@+id/numberTextView"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_gravity="right|center_vertical"
    android:labelFor="@+id/numberEditText"
    android:text="@string/number"
    android:textAppearance="?android:attr/textAppearanceMedium" />
<EditText
    android:id="@+id/numberEditText"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_row="1"
    android:digits="0123456789"
    android:inputType="number"
    android:maxLength="3" />

<requestFocus />
```

   <TextView
        android:id="@+id/numberDisplayTextView"
        android:layout_width="50dp"
        android:layout_height="wrap_content"
        android:layout_column="1"
        android:layout_gravity="fill_horizontal"
        android:layout_row="1"

```
    android:padding="@dimen/textview_padding"
    android:textAppearance="?android:attr/textAppearanceMedium" />



<TextView
    android:id="@+id/customPercentTextView"
    android:layout_width="wrap_content"
    android:layout_gravity="right|center_vertical"
    android:text="@string/custom_tip_percentage"
    android:textAppearance="?android:attr/textAppearanceMedium" />

<SeekBar
    android:id="@+id/customTipSeekBar"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_gravity="fill_horizontal"
    android:max="30"
    android:progress="18" />

<LinearLayout
    android:id="@+id/percentLinearLayout"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_gravity="fill_horizontal" >

    <TextView
        android:id="@+id/percent15TextView"
        android:layout_width="wrap_content"
        android:layout_height="match_parent"
        android:layout_weight="1"
        android:gravity="center"
        android:text="@string/fifteen_percent"
        android:textAppearance="?android:attr/textAppearanceMedium" />

    <TextView
        android:id="@+id/percentCustomTextView"
        android:layout_width="wrap_content"
        android:layout_height="match_parent"
        android:layout_weight="1"
        android:gravity="center"
        android:text="@string/eighteen_percent"
        android:textAppearance="?android:attr/textAppearanceMedium" />

</LinearLayout>

<TextView
    android:id="@+id/tipTextView"
    android:layout_gravity="right|center_vertical"
    android:text="@string/tip"
    android:textAppearance="?android:attr/textAppearanceMedium"  />

<LinearLayout
    android:id="@+id/tipLinearLayout"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_gravity="fill_horizontal" >

    <TextView
        android:id="@+id/tip15TextView"
        android:layout_width="0dp"
        android:layout_height="match_parent"
        android:layout_gravity="center"
        android:layout_marginRight="@dimen/textview_margin"
        android:layout_weight="1"
        android:background="@android:color/holo_orange_light"
        android:gravity="center"
        android:padding="@dimen/textview_padding"
        android:textAppearance="?android:attr/textAppearanceMedium" />

    <TextView
        android:id="@+id/tipCustomTextView"
        android:layout_width="0dp"
        android:layout_height="match_parent"
        android:layout_gravity="center"
        android:layout_weight="1"
        android:background="@android:color/holo_orange_light"
        android:gravity="center"
        android:padding="@dimen/textview_padding"
        android:textAppearance="?android:attr/textAppearanceMedium" />

</LinearLayout>

<TextView
    android:id="@+id/totalTextView"
    android:layout_gravity="right|center_vertical"
    android:text="@string/total"
    android:textAppearance="?android:attr/textAppearanceMedium" />

<LinearLayout
    android:id="@+id/totalLinearLayout"
    android:layout_height="wrap_content"
    android:layout_column="1"
    android:layout_gravity="fill_horizontal" >

    <TextView
        android:id="@+id/total15TextView"
        android:layout_width="0dp"
        android:layout_height="match_parent"
        android:layout_gravity="center"
        android:layout_marginRight="@dimen/textview_margin"
        android:layout_weight="1"
        android:background="@android:color/holo_orange_light"
        android:gravity="center"
        android:padding="@dimen/textview_padding"
        android:textAppearance="?android:attr/textAppearanceMedium" />

    <TextView
        android:id="@+id/totalCustomTextView"
        android:layout_width="0dp"
        android:layout_height="match_parent"
        android:layout_gravity="center"
        android:layout_weight="1"
        android:background="@android:color/holo_orange_light"
        android:gravity="center"
        android:padding="@dimen/textview_padding"
        android:textAppearance="?android:attr/textAppearanceMedium" />

</LinearLayout>

<Space />
```

</GridLayout>

Comment (abdoulkaled) [inline ok:1]: package com.example.gakasgado.tip_calculator;

import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;// for saving state information
import android.view.Menu;
import android.view.MenuItem;
import java.text.NumberFormat; // for currency formatting

import android.app.Activity; // base class for activities
import android.text.Editable; // for EditText event handling
import android.text.TextWatcher; // EditText listener
import android.widget.EditText; // for bill amount input
import android.widget.SeekBar; // for changing custom tip percentage
import android.widget.SeekBar.OnSeekBarChangeListener; // SeekBar listener
import android.widget.TextView; // for displaying text

// MainActivity class for the Tip Calculator app
public class MainActivity extends Activity
{
    // currency and percent formatters
    private static final NumberFormat currencyFormat =
            NumberFormat.getCurrencyInstance();
    private static final NumberFormat percentFormat =
            NumberFormat.getPercentInstance();

```
private double billAmount = 0.0; // bill amount entered by the user
private int num=1; //number in party entered by the user
private double customPercent = 0.18; // initial custom tip percentage
private TextView amountDisplayTextView; // shows formatted bill amount
private TextView numberDisplayTextView;
private TextView percentCustomTextView; // shows custom tip percentage
private TextView tip15TextView; // shows 15% tip
private TextView total15TextView; // shows total with 15% tip
private TextView tipCustomTextView; // shows custom tip amount
private TextView totalCustomTextView; // shows total with custom tip

// called when the activity is first created
@Override
protected void onCreate(Bundle savedInstanceState)
{
    super.onCreate(savedInstanceState); // call superclass's version
    setContentView(R.layout.activity_main); // inflate the GUI

    // get references to the TextViews
    // that MainActivity interacts with programmatically
    amountDisplayTextView =
            (TextView) findViewById(R.id.amountDisplayTextView);
    numberDisplayTextView = (TextView) findViewById(R.id.numberDisplayTextView);

    percentCustomTextView =
            (TextView) findViewById(R.id.percentCustomTextView);
    tip15TextView = (TextView) findViewById(R.id.tip15TextView);
    total15TextView = (TextView) findViewById(R.id.total15TextView);
    tipCustomTextView = (TextView) findViewById(R.id.tipCustomTextView);
    totalCustomTextView =
            (TextView) findViewById(R.id.totalCustomTextView);

    // update GUI based on billAmount and customPercent
    amountDisplayTextView.setText(
            currencyFormat.format(billAmount));
    numberDisplayTextView.setInputType(num);
    updateStandard(); // update the 15% tip TextViews
    updateCustom(); // update the custom tip TextViews

    // set amountEditText's TextWatcher
    EditText amountEditText =
            (EditText) findViewById(R.id.amountEditText);
    amountEditText.addTextChangedListener(amountEditTextWatcher);

    EditText numberEditText = (EditText) findViewById(R.id.numberEditText);
    numberEditText.addTextChangedListener(numberEditTextWatcher);

    // set customTipSeekBar's OnSeekBarChangeListener
    SeekBar customTipSeekBar =
            (SeekBar) findViewById(R.id.customTipSeekBar);
    customTipSeekBar.setOnSeekBarChangeListener(customSeekBarListener);
} // end method onCreate

// updates 15% tip TextViews
private void updateStandard()
{
    // calculate 15% tip and total
    double fifteenPercentTip = billAmount * 0.15;
    double fifteenPercentTotal = billAmount + fifteenPercentTip;

    fifteenPercentTotal= fifteenPercentTotal/num;

    // display 15% tip and total formatted as currency
    tip15TextView.setText(currencyFormat.format(fifteenPercentTip));
    total15TextView.setText(currencyFormat.format(fifteenPercentTotal));
} // end method updateStandard

// updates the custom tip and total TextViews
private void updateCustom()
{
    // show customPercent in percentCustomTextView formatted as %
    percentCustomTextView.setText(percentFormat.format(customPercent));

    // calculate the custom tip and total
    double customTip = billAmount * customPercent;
    double customTotal = billAmount + customTip;

    //calculate the total custom tip by person

     customTotal= customTotal/ num;

    // display custom tip and total formatted as currency
    tipCustomTextView.setText(currencyFormat.format(customTip));
    totalCustomTextView.setText(currencyFormat.format(customTotal));

} // end method updateCustom

// called when the user changes the position of SeekBar
private OnSeekBarChangeListener customSeekBarListener =
        new OnSeekBarChangeListener()
        {
            // update customPercent, then call updateCustom
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress,
                                          boolean fromUser)
            {
                // sets customPercent to position of the SeekBar's thumb
                customPercent = progress / 100.0;
                updateCustom(); // update the custom tip TextViews
            } // end method onProgressChanged

            @Override
            public void onStartTrackingTouch(SeekBar seekBar)
            {
            } // end method onStartTrackingTouch

            @Override
            public void onStopTrackingTouch(SeekBar seekBar)
            {
            } // end method onStopTrackingTouch
        }; // end OnSeekBarChangeListener

// event-handling object that responds to amountEditText's events
private TextWatcher amountEditTextWatcher = new TextWatcher()
{
    // called when the user enters a number
    @Override
    public void onTextChanged(CharSequence s, int start,
                              int before, int count)
    {
        // convert amountEditText's text to a double
        try
        {
            billAmount = Double.parseDouble(s.toString()) / 100.0;
        } // end try
        catch (NumberFormatException e)
        {
            billAmount = 0.0; // default if an exception occurs
        } // end catch

        // display currency formatted bill amount
        amountDisplayTextView.setText(currencyFormat.format(billAmount));
        updateStandard(); // update the 15% tip TextViews
        updateCustom(); // update the custom tip TextViews
    } // end method onTextChanged

    @Override
    public void afterTextChanged(Editable s)
    {
    } // end method afterTextChanged

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count,
                                  int after)
    {
    } // end method beforeTextChanged
}; // end amountEditTextWatcher

private TextWatcher numberEditTextWatcher = new TextWatcher()
{
    // called when the user enters a number
    @Override
    public void onTextChanged(CharSequence s, int start,
                              int before, int count)
    {
        // convert numberEditText's text to a int

        try
        {

            num= Integer.parseInt(s.toString());
        } // end try
        catch (NumberFormatException e)
        {
            num = 1; // default if an exception occurs
        } // end catch

        // display currency formatted number
       numberDisplayTextView.setInputType(num);

        updateStandard(); //update the 15% tip TextViews
        updateCustom(); //update the custom tip TextViews
    } // end method onTextChanged

    @Override
    public void afterTextChanged(Editable s)
    {
    } // end method afterTextChanged

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count,
                                  int after)
    {
    } // end method beforeTextChanged
}; // end numberEditTextWatcher
```

} // end class MainActivity

**Comment by Bansenauer-Cascadia** [comment]
# Code Review

**Complete and correct code to:**

---
- add a text box to enter the party size 
- account for invalid user entries 
- update the total when the party size text is changed (event handler)

**Incorrect or missing code to:**

---
- include an additional strings.xml file for Chinese localization
- calculate the amount owed per person (divide by zero check)

**Pull request closed without merging**